internal/backends/nodejs/nodejs.go

```diff
@@ -246,6 +247,19 @@ func nodejsGuess() (map[api.PkgName]bool, bool) {
 	return pkgs, true
 }
 
+func parseYarnSpec(specBytes []byte) map[api.PkgName]api.PkgVersion {
+	contents := string(specBytes)
+	r := regexp.MustCompile(`(?m)^"?([^@ \n]+).+:\n  version "(.+)"$`)
```

Comment:
Probably better to use the built-in json parser https://gobyexample.com/json

Reply:
the yarn lockfile format isn't json (nor yaml), so that won't work. the regex is correct